configs/resource.go

```diff
@@ -111,13 +111,15 @@ func decodeResourceBlock(block *hcl.Block) (*Resource, hcl.Diagnostics) {
 
 	if attr, exists := content.Attributes["for_each"]; exists {
 		r.ForEach = attr.Expr
-		// We currently parse this, but don't yet do anything with it.
-		diags = append(diags, &hcl.Diagnostic{
-			Severity: hcl.DiagError,
-			Summary:  "Reserved argument name in resource block",
-			Detail:   fmt.Sprintf("The name %q is reserved for use in a future version of Terraform.", attr.Name),
-			Subject:  &attr.NameRange,
-		})
+		// Cannot have count and for_each on the same resource block
+		if r.Count != nil {
+			diags = append(diags, &hcl.Diagnostic{
+				Severity: hcl.DiagError,
+				Summary:  "Invalid config, cannot use count and for_each in same block",
```

Comment:
The sentence fragment here made me trip over reading this sentence a bit. :confounded: 

Most of our "modern" error messages try to start with a concise statement of the broad problem as the summary and then elaborate more in the detail. Since this is a situation where it's the combination of two things that is invalid rather than one thing, the usual "Invalid <whatever it is>" summary won't work here, but maybe something like this?

```
Error: Invalid combination of "count" and "for_each"

  <source snippet>

The "count" and "for_each" meta-arguments are mutually-exclusive, because each one chooses
a different identifier type for the resource instances.
```

(This term "meta-arguments" has always felt a little weird to me, but that's what it's called throughout the documentation and so hopefully using that terminology in the error message will help users find further documentation about these features if they are unsure how to proceed.)